src/components/widget-cta.js

```diff
@@ -0,0 +1,153 @@
+// If we need cover support for cta widget later,
+// don't add it here.
+
+import React, {createElement as el} from 'react'
+import Checkbox from 'rebass/dist/Checkbox'
+import ButtonOutline from 'rebass/dist/ButtonOutline'
+
+import TextareaAutosize from './textarea-autosize'
+import {isUrlOrBlank} from '../util/url'
+import {widgetLeftStyle} from './rebass-theme'
+
+// Gets src or href from iframe or a
+// http://www.regexpal.com/ test string:
+// asfd  dfadf ads <iframe  a="ffff" src="fooo" sss></iframe> <a hRef='http://,,,'>thing!</a> asdf
+const extractLink = /<(iframe|a)\s+[^>]*(?:src|href)=["'](\S+)["'][^>]*>/i
```

Comment:
In theory (so this is mostly edge cases and super edge cases);

- iirc there can be spaces between `<` and the tag name
- you may be matching `src` that's part of a different payload or even custom attribute (`<iframe foosrc="..">` or `<iframe onclick="foo.src='bar';">`) 
- the value may not be quoted (`<iframe src=foo.html>')
- there can be whitespace next to the equal sign (I think even before, but certainly after) that may be ignored (stuff like `<iframe src= "foo">` and `<iframe src ="foo">` (?) and even `<iframe src= foo>` iirc). The spec is weird like that.
- the `(\S+)` should be some kind of backreference to the quote parsed earlier, but at least have a `?` to be non-greedy otherwise it will skip quotes as long as another quote can be found (`<iframe src="foo" and="bar">` would lead to `foo" and ="bar`), so use `(\S+?)` to make the `+` be "non-greedy".

I hope `extractLink` is sanitized afterwards because there can be a lot of crap inside of it. But that's beyond the scope of the question or the PR.